lib/travis/build/script/matlab.rb

```diff
@@ -0,0 +1,46 @@
+module Travis
+  module Build
+    class Script
+      class Matlab < Script
+        MATLAB_INSTALLER_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/ephemeral-matlab/v0/install.sh'.freeze
+        MATLAB_DEPS_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/matlab-deps/v0/install.sh'.freeze
+        MATLAB_START = 'matlab -batch'.freeze
+        MATLAB_COMMAND = "assertSuccess(runtests('IncludeSubfolders',true));".freeze
+
+        DEFAULTS = {
+          matlab: 'latest'
+        }.freeze
+
+        def export
+          super
+          sh.export 'TRAVIS_MATLAB_VERSION', release.shellescape,
+                    echo: false
+        end
+
+        def setup
```

Comment:
Another approach would be to roll in this validation into either `install.sh` script you have better control over and error the build here.

Reply:
Indeed that's actually what we're doing inside our `install.sh` script 😄

Reply:
How would one propose changes to `install.sh`? How would one troubleshoot issues with it if such a need arises? Is it open sourced somewhere?

Reply:
I think people would need to go through the "community supported" channel of the language and send an email to continuous-integration@mathworks.com to propose changes. Currently, other than the readable shell script in install.sh, the application that does the work of providing access to  MATLAB isn't in a public repo.

Reply:
Understood. Could this bit of information added to the build log output and the documentation? For example:

https://github.com/travis-ci/travis-build/blob/24ca6b8fbb931756d5a9502490db0aa82ab52053/lib/travis/build/script/d.rb#L21-L28

(It might be appropriate to say that Mathworks provides support, instead of individuals; it's up to you.)